lms/views/api/exceptions.py

```diff
@@ -26,18 +27,28 @@ class ExceptionViews:
     Error responses from the API have JSON bodies with the following keys (all
     optional):
 
-    1. "message": An error message for the frontend to show to the user.
+    1. "error_code": A unique string that identifies the error to the frontend.
+
+       If "error_code" is present the frontend should show an error dialog
+       relevant to the particular error code. For example if "error_code" is
+       "canvas_api_permission_error" the frontend should show a
+       "You don't have permission" error dialog. The error dialog should
+       include a [Try again] button that opens the "canvas_api.oauth.authorize"
```

Comment:
Hmm, this comment implies different handling of the error than I have implemented in my PR. In my PR I assumed that a permission error indicates that the access token is fine and so there is no point in repeating the authorization part of the process. Can you think of a situation where `canvas_api_permission_error` will be returned but re-authorization will help?

Reply:
:man_shrugging: Who knows? The general approach in the LMS app so far has been that if anything goes wrong with a Canvas API request, the <kbd>Try again</kbd> button re-starts the _whole_ process, beginning with getting a new access token, regardless of whether we think a new access token is necessary or not. This re-try-from-the-very-start approach gives <kbd>Try again</kbd> the best possible chance of succeeding, keeps the code simpler (no need to distinguish re-tries that do and don't require an access token), and avoids potential issues where our code thinks a new access token isn't required but in fact one is (and so the user is stuck in a loop due to our faulty logic)